File: unidbg-api/src/main/java/com/vortexdbg/spi/LibraryFile.java
package com.vortexdbg.spi;

import com.vortexdbg.Emulator;

import java.io.IOException;
import java.nio.ByteBuffer;

public interface LibraryFile {

    String getName();

    String getMapRegionName();

    LibraryFile resolveLibrary(Emulator<?> emulator, String soName) throws IOException;

    ByteBuffer mapBuffer() throws IOException;

    String getPath();

    long getFileSize();

}

File: vortexdbg-api/src/main/kotlin/com/vortexdbg/spi/LibraryFile.kt
package com.vortexdbg.spi

import com.vortexdbg.Emulator

import java.io.IOException
import java.nio.ByteBuffer

interface LibraryFile {

    fun getName(): String

    fun getMapRegionName(): String

    @Throws(IOException::class)
    fun resolveLibrary(emulator: Emulator<*>, soName: String): LibraryFile?

    @Throws(IOException::class)
    fun mapBuffer(): ByteBuffer

    fun getPath(): String

    fun getFileSize(): Long

}
